sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -82,21 +79,19 @@ class KernelProgramCache {
     return {MKernelsPerProgramCache, MKernelsPerProgramCacheMutex};
   }
 
-  template <class Predicate> void waitUntilBuilt(Predicate Pred) const {
-    std::unique_lock<std::mutex> Lock(MBuildCVMutex);
-
-    MBuildCV.wait(Lock, Pred);
+  template <class Predicate>
+  void waitUntilBuilt(std::condition_variable &CV,
+                      std::unique_lock<std::mutex> &Lock,
+                      Predicate Pred) const {
+    CV.wait(Lock, Pred);
   }
```

Comment:
I believe this function should have a comment, saying that `condition_variable` and `unique_lock` should be for the same `BuildResult`.